### DISCUSSION TITLE
Build for i586

### DISCUSSION THREAD

**Discussion by alexanderzhirov** [dlang/dmd#14339] [General]
I managed to [build](https://github.com/ldc-developers/ldc/issues/4013#issuecomment-1199125440) `ldc2` with libraries for the `i586` architecture. Is it possible to build libraries for `DMD` for the `i586` architecture? I manage to build `DMD` for `i586`, but the libraries are built for `i686` (`runtime` and `phobos`).

What tool can I use to rebuild libraries to specify `i586`? I tried adding `DFLAGS` `-mcpu=native` and `CFLAGS` `-march=i586` to the `posix.mak` source code when building `DMD` on an `i586` PC, but this did not work during assembly.

**Comment by PetarKirov** [comment]
The dmd backend has historically supported even older targets, but as far as I can see the frontend currently does not have a way to configure it:

https://github.com/dlang/dmd/blob/043caf10136bc37b19c5c7373e0b4bd324c08055/compiler/src/dmd/target.d#L30-L47

So, in theory it should be possible, but it's unknown at the moment how much effort is involved.

If you do have another development machine, I would suggest developing there and then using LDC to cross-compile to i586 for deployment.

**Reply by alexanderzhirov** [reply]
In fact, I do this because I initially experienced a problem with building `DMD` for the `i586` architecture.
